### ISSUE TITLE
Images being used with .predict() don't appear to be normalized the same way as the images pulled using get_x from the datasets (for feeding into predict_array).

### ISSUE THREAD

**Issue by nwertzberger** [fastai/fastai#535]
Hello,

I've been trying to track down why I get a different answer for:

    learner.predict_array(dataset.val_dl.dataset.get_x(0)[None])

vs what i see in preds from running (learner.predict()) using the same dataset.  I noticed the following:

* When I look up predict_with_targs, it seems that it turns the dataloader into an iterator, and then shoves those values into the model via this:

        def predict_with_targs(m, dl):
            m.eval()
            if hasattr(m, 'reset'): m.reset()
            res = []
            for *x,y in iter(dl): res.append([get_prediction(m(*VV(x))),y])
            preda,targa = zip(*res)
            return to_np(torch.cat(preda)), to_np(torch.cat(targa))

* When I pull out some data using this method, I notice that the resulting tensor is NOT normalized, but the dimensions are 3x128x128 (what my model expects):

        x = list(iter(data2.val_dl))[0]
        to_np(x[0][0,:,:,:])
        # This shows a tensor with lots of values bigger than 1 and smaller than -1

* When I pull from the dataset using get_x, it IS normalized, but the dimensions are 128x128x3

Given that the other interfaces return normalized images, i think this might not be on purpose, but I'm no expert on what should or should not be normalized. I know that it not being normalized makes lime integration difficult, and, because the dimensions of the images pulled in each case is different, I have to transpose them as well, which seems awkward.

**Comment by jph00** [comment]
You should never call `get_x` directly. Instead just use the standard dataset indexer: `dataset.val_ds[i]`

**Comment by nwertzberger** [comment]
Is it intentional that the resulting tensor is not normalized between 0 and
1?

**Comment by yanneta** [comment]
Here is the answer to your question

https://discuss.pytorch.org/t/whats-the-range-of-the-input-value-desired-to-use-pretrained-resnet152-and-vgg19/1683

**Comment by nwertzberger** [comment]
I’m sorry, I am not quite understanding the answer here. In one interface,
I see the images are normalized from 0 to 1, and that makes sense. I expect
to see normalized images as mentioned In Torchvision. On the iterator
interface, I see tensors that do not appear to be normalized. Also, when I
give a normalized image to my network, it gives significantly different
predictions than my non normalized input, making me think that nothing is
auto-normalizing the input, at least through predict_array.
